lib/modules/datasource/terraform-provider/index.ts

```diff
@@ -282,6 +282,44 @@ export class TerraformProviderDatasource extends TerraformDatasource {
     return filteredResult.length === result.length ? filteredResult : null;
   }
 
+  @cache({
+    namespace: `datasource-${TerraformProviderDatasource.id}-zip-hashes`,
+    key: (registryURL: string, repository: string, version: string) =>
+      `${registryURL}/${repository}/${version}`,
+  })
+  async getZipHashes(builds: TerraformBuild[]): Promise<string[] | undefined> {
+    if (builds.length === 0) {
+      return [];
+    }
+
+    const zipHashUrl = builds[0].shasums_url;
+
+    if (!zipHashUrl) {
+      return [];
+    }
+
+    // The hashes are formatted as the result of sha256sum in plain text, each line: <hash>\t<filename>
+    let rawHashData;
+    try {
+      rawHashData = (await this.http.get(zipHashUrl)).body;
+    } catch (err) {
+      /* istanbul ignore next */
+      if (err instanceof ExternalHostError) {
+        throw err;
+      }
+      logger.debug(
+        { err, zipHashUrl },
+        `Failed to retrieve zip hashes from ${zipHashUrl}`
+      );
+      return undefined;
```

Comment:
This is so, that in this case the result is not cached.

Reply:
Seems I should leave it as `undefined` to make sure it does not replace the cached object in case of failure.
https://github.com/renovatebot/renovate/blob/95076df302d4791b339d9798ce3f15dfd18f0ed4/lib/util/cache/package/decorator.ts#L122

This return value is not saying there is no data, just that we don't know at all. It will only happen on `ExternalHostError`, otherwise (for a syntactically ok HTTP error like a 404) `[]` is returned (and cached).